compiler/rustc_passes/src/loops.rs

```diff
@@ -216,6 +228,22 @@ impl<'hir> Visitor<'hir> for CheckLoopVisitor<'hir> {
                     Err(hir::LoopIdError::UnresolvedLabel) => None,
                 };
 
+                // a #[const_continue] must be to a block that participates in #[loop_match]
+                let attrs = self.tcx.hir_attrs(e.hir_id);
+                if attrs.iter().any(|attr| attr.has_name(sym::const_continue)) {
+                    if let Some(break_label) = break_label.label {
+                        let is_target_label = |cx: &Context| match cx {
+                            Context::LoopMatch { labeled_block } => break_label == *labeled_block,
+                            _ => false,
+                        };
+
+                        if !self.cx_stack.iter().rev().any(is_target_label) {
+                            let span = break_label.ident.span;
+                            self.tcx.dcx().emit_fatal(ConstContinueBadLabel { span });
```

Comment:
none of the other errors in this file are `emit_fatal`, but without it the error is not emitted in time and we run into `bug!` down the line.

Reply:
Would returning from this function after `emit_error` here work?

Reply:
no, that still ICEs unfortunately